Norm_end_to_end.Rmd

```diff
@@ -432,200 +524,450 @@ colnames(study_summary) <- colnames(study_sample)
+
+par(mfrow = c(1, 2), mar = c(4.5, 4.5, 4.5, 1))
+plot_pca(pca_res, pc_x = 1, pc_y = 2, pch = 21,
+         col = col_sample, bg = paste0(col_sample, 80),
+         labels = res$injection_index, main = "NOMIS")
+legend("topleft", inset = c(0, -0.24), col = col_phenotype,
+       legend = names(col_phenotype), lty=1, lwd = 2, xpd = TRUE, ncol = 3,
+       cex = 0.7, title = "Phenotype", title.cex = 0.8)
+plot_pca(pca_res, pc_x = 3, pc_y = 4, pch = 21,
+         col = col_sample, bg = paste0(col_sample, 80),
+         labels = res$injection_index)
+```
+
+
+
+### Apply to dataset
+
+```{r between-sample-choose-best, eval=FALSE, include=TRUE}
 # remove the assay or just rewrite code better as to not actually put the assay
-# in the SumExp object 
+# in the SumExp object
 
-#' create another assay norm removing the filled data. 
-tmp <- # put here 
+#' create another assay norm removing the filled data.
+tmp <- # put here
 tmp[is.na(assay(res_pos, "raw"))] <- NA
 assays(res_pos)$norm_nofill <- tmp
 rm(tmp)
 
 #' once settled on one method, i will change code before and after to match this
 ```
 
-## Within batch normalization 
+## Within batch normalization
 
 - remove injection order dependent signal drift
 fit feature-wise linear model (y ~ inj_idx) to the (log2 transformed) abundance
 data of the QC samples within each batch and comapre the slopes for each feature
-between the batches. 
+between the batches.
 
 Because the Gap filling evaluation showed quite a significant difference in
 signal between detected values compared to filled in (especially at low signal).
-therefore we will used non-filled values for this next step. 
+therefore we will used non-filled values for this next step.
 
 Flags: remove fitted linear models for features for which valid
 measurement do not span at least 2/3 of the inject index range or for which the
-mean of the absolute residual is larger than 0.5. 
+mean of the absolute residual is larger than 0.5.
 
  - look for highest abs slope plot
 
 ### linear model
 
 ```{r}
 #' take the log2 of the between sample norm results BUT remove filled in data
-tmp <- log2(assay(res, "median_filled_500")) 
+tmp <- log2(assay(res, "median_filled_500"))
 tmp[is.na(assay(res, "raw"))] <- NA
 
-#' Fit linear model y ~ inj_idx
-model_lm <- xcms:::rowFitModel(
-    y ~ inj_idx, 
-    data = data.frame(inj_idx = 
-        res$injection_index[res$phenotype == "QC"]), 
-    y = tmp[, res$phenotype == "QC"], 
-    method = "lm", 
+#' Fit linear model to QC samples only.
+qc_lm <- xcms:::rowFitModel(
+    y ~ injection_index,
+    data = as.data.frame(colData(res)[res$phenotype == "QC", ]),
+    y = tmp[, res$phenotype == "QC"],
+    method = "lm",
     minVals = 3
 )
 
-##' Calculate slopes
-slps_lm <- vapply(model_lm, function(z) {
+#' Fit linear model to study samples only. Requires injection index to
+#' be independent of any experimental covariate.
+study_lm <- xcms:::rowFitModel(
```

Comment:
this is more for internal evaluation. assuming that QC samples are a pool of all study samples and *behave* the same way as study samples **and** that study samples are properly randomized, we would expect to see a similar intensity drift for QC and for study samples. In theory. For this data set that is the case for some, but not all features. and the differences actually worry me a lot. they make me think that maybe it's not a good idea to adjust the study samples by injection order dependent signal drift estimated on QC samples...

Reply:
Yes we can discuss it during a little "end-to-end strategy" meeting. But as we already discussed any adjustment on the study sample using these QC might not be the best idea... In my opinion it might be best to stick to between sample norm and keep within-batch strategy for dataset on which it is actually applicable.

Reply:
totally agree! for the end to end we should drop it - and at maximum mention that the possibility exists.